### PR TITLE
Add pty to unsafeimports

### PR DESCRIPTION
**Pull request by ajohnston9**
The following code produces a pickle file that fickling fails to detect as malicious:

```
import pickle
import pickletools

payload = b'''(cpty\nspawn\nS"id"\no.'''
pickletools.dis(payload, annotate=1)

with open('pwn.pkl', 'wb') as f:
    f.write(payload)
```

This is because this technique uses `pty` and does not leave _var0 unused. This PR adds `pty` to `unsafe_imports` as a quick fix to ensure the primitive behind this technique is detected.

### CODE REVIEW

**Review by ESultanik — APPROVED** [review]
This is great, thanks! It'd be nice to have a unit test to confirm that we now catch this attack, but we can add that after this PR is merged. Once you sign the CLA we can merge.

**Comment by ajohnston9** [comment]
I've added a relevant test and signed the CLA! Let me know if there's anything else I can do.

File: test/test_hook.py [hunk 1]
```diff
@@ -39,13 +39,19 @@ def __reduce__(self):
...
 
         # Save the payload in a pickle file
         with open("unsafe.pickle", "wb") as f:
             pickle.dump(payload, f)
+        with open("unsafe_pty.pickle", "wb") as f:
+            f.write(payload_pty)
 
         try:
             numpy.load("unsafe.pickle", allow_pickle=True)
+            numpy.load("unsafe_pty.pickle", allow_pickle=True)
```
Comment (woodruffw) [inline test/test_hook.py:54]: Might be missing something, but shouldn't this be in its own test/`try-except` block at a minimum? I believe as-is the first line will always trip the exception, so this will never be tested.

**Comment by thomas-chauchefoin-tob** [comment]
@ajohnston9 Commit https://github.com/trailofbits/fickling/pull/108/commits/c202961fca96fa61da1bc14f725afc44aa02ef78 was done with an email address that is not associated with the GitHub account you used to sign the CLA, can you look into it? I'll take care of the rest.
